common/json/impl/boost/boost_json.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef BOOST_JSON_HPP
```

Comment:
This actually conflicts with a definition in `Boost`. Should we prefix all our defs with `MENDER`?

Reply:
Generally the naming convention is not great. If you just compile this into a library, it will be called `libboost_json` which is again a conflict with Boost. And the `common/MODULE/impl/BACKEND/BACKEND_MODULE.[ch]pp` scheme is very redundant, IMHO. `src.[ch]pp` or `impl.[ch]pp` would make sense to me just fine. And the guard macro could be `JSON_BOOST_WHATEVER` or `MEN_JSON_BOOST_WHATEVER`.

Reply:
Yes, agree fully now that I have played around with this a bit!

Reply:
@vpodzime should we not fix this?